test/e2e/framework/deployment.go

```diff
@@ -77,8 +79,11 @@ func (f *Framework) NewEchoDeployment(opts ...func(*deploymentOptions)) {
 	for _, o := range opts {
 		o(options)
 	}
-
-	deployment := newDeployment(options.name, options.namespace, "k8s.gcr.io/ingress-nginx/e2e-test-echo@sha256:131ece0637b29231470cfaa04690c2966a2e0b147d3c9df080a0857b78982410", 80, int32(options.replicas),
+	e2eImage := os.Getenv("E2E_BASE_IMAGE")
+	if e2eImage == "" {
+		assert.Nil(ginkgo.GinkgoT(), errors.New("E2E_BASE_IMAGE was not defined"), "creating e2e base image")
```

Comment:
I'm not sure what this is doing, probably you just want to change the function signature and return an error? (or do a fatal here)